### ISSUE TITLE
Create Documentation, Guides and API

### ISSUE THREAD

**Issue by faustinoaq** [amberframework/amberframework.org#57]
Hi @amberframework/contributors 

Current guides are a bit confusing for some people. Currently they are merged with documentation, recipes and some api.

I think we can divide Guides and Documentation publishing guides on https://amberframework.org/guides and documentation on Gitbook, Also we should publish an API page

So:

- [ ] https://amberframework.org/guides - Step by step guide like [Elixir](https://elixir-lang.org/getting-started/introduction.html) does
- [ ] https://amberframework.org/docs - We can cleanup [Gitbook](https://docs.amberframework.org/) and use it for documentation only, stuff like what are the framework components and deep explanation in amber project structure.
- [ ] https://amberframework.org/api - We can use `crystal docs` documenting Amber codebase following current [guidelines](https://crystal-lang.org/docs/conventions/documenting_code.html)

These 3 steps should allow us to remove [this](https://github.com/amberframework/amberframework.org/blob/master/src/helpers/guide_helper.cr) hack, and simplify the huge [sidebar](https://github.com/amberframework/amberframework.org/blob/master/src/views/home/_sidebar.slang) written by hand.

**Comment by faustinoaq** [comment]
Also would solve #53

**Comment by jpotts18** [comment]
@faustinoaq I've taken a first pass at an outline for the docs. I basically stole it from [rails guides started](http://guides.rubyonrails.org)

## Guide
### Chapter 1 - Start Here:
- Guide Assumptions
  - Installing Crystal
  - Dependency Manager
  - SQLite Database
- What is Amber?
  - Manifest
  - Convention over Config
- Creating a New Amber Project	
  - Installing Amber
  - Creating the Blog App
- Hello, Amber!
  - Starting the Web Server
  - Say Hello to Amber
  - Setting the Homepage (routes)
- Getting Up and Running
  - First Controller (Articles#new)
  - First Form
  - Articles#create
  - First Model Articles
  - First Migration
  - Save Article
  - Articles#show
  - Articles#index
  - link_to intro
  - Article Validation
  - Articles#update
  - View partials to clean up duplication
  - Articles#delete
- Adding a Second Model
   - Comments
   - Associating Models
   - Routing
   - Generate Controller
- Refactoring
- Deleting Comments
- Security
- What's Next?
- Common Gotchas
### Chapter 2 - Models:
- ORM Basics
- ORM Migrations
- ORM Validations
- ORM Callbacks
- ORM Associations
- ORM Query Interface
### Chapter 3 - Views:
- Templates (Slang)
- Layouts
- Partials
### Chapter 4 - Controllers
- Methods
- Actions
- Parameters
- Session
- Cookies
- Formats (JSON / XML)
- Filters
- Request
- Response

**Comment by marksiemers** [comment]
To confirm and/or clarify:

- Guides will be like tutorials and/or cookbooks
  - Self-hosted and maintained (perhaps with a database-backed system)?
- Docs will be like what Rails calls "Guides" and based on the outline by @jpotts18 
   - Hosted and maintained in Gitbook?
- API will be the documentation generated by `crystal docs`
   - Self-hosted? Kept in a separate repo from `amberframework/amber`

Let's all get on the same page about the details of the end goal so we can make a coordinated effort toward that goal.

**Comment by faustinoaq** [comment]
> Guides will be like tutorials and/or cookbooks
>> Self-hosted and maintained (perhaps with a database-backed system)?

Yes , self hosted, but I rather prefer a git-based guides, is easier to maintain and collaborate.

> Docs will be like what Rails calls "Guides" and based on the outline by @jpotts18
>> Hosted and maintained in Gitbook?

Yeah, I think we can use Gitbook for more detailed dpcumentation.

> API will be the documentation generated by crystal docs
>> Self-hosted? Kept in a separate repo from amberframework/amber

Yes, generated by `crystal docs` and I think we can make some changes to make it more amberish

Good goals for now, However I think would be awesome to have all in a same place, just like hexdocs does.

By example: Phoenix (one of the framework amber is based) uses only PAGES (aka guides) and MODULES (aka. API documentation)

## Phoenix PAGES (Guides)

![screenshot_20171204_184125](https://user-images.githubusercontent.com/3067335/33582386-fba52270-d922-11e7-881e-017b639cf79b.png)

## Phoenix MODULES (API documentation)

![screenshot_20171204_184224](https://user-images.githubusercontent.com/3067335/33582423-2e882bf6-d923-11e7-9d97-bcdb1d41979e.png)

Maybe could be possible refactoring `crystal docs` allowing to add custom Markdown files to sidebar, WDTY @straight-shoota?

We could even storage https://crystal-lang.org/docs/ and https://crystal-lang.org/api/ in a same place like Elixir.

Perhaps after https://github.com/crystal-lang/crystal/pull/4755 and https://github.com/crystal-lang/crystal/pull/5229,  that could be the next step 😅

**Comment by straight-shoota** [comment]
It would certainly be great to integrate API docs and other documentation resources more tightly. But the exact needs will certainly differ between different projects. So, the docs generator could perhaps provide a means to inject content (like sidebar links) and allow to apply certain customizations.
There has also been the idea of reducing the docs generator into a compiler feature that *extracts* documentation data from source. The outputs in a standard format (JSON) can then be used to generate different documentation formats (HTML, PDF, ebook whatever) and allows projects to customize the documentation to their needs in detail.

**Comment by faustinoaq** [comment]
We're using https://amberframework.gitbook.io/amber/getting-started now :tada: 

An Amber API page is still missing, though
